parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java

```diff
@@ -65,28 +68,25 @@
   private ParquetUtil() {
   }
 
-  // Access modifier is package-private, to only allow use from existing tests
-  static Metrics fileMetrics(InputFile file) {
-    return fileMetrics(file, MetricsConfig.getDefault());
-  }
-
   public static Metrics fileMetrics(InputFile file, MetricsConfig metricsConfig) {
     return fileMetrics(file, metricsConfig, null);
   }
 
   public static Metrics fileMetrics(InputFile file, MetricsConfig metricsConfig, NameMapping nameMapping) {
     try (ParquetFileReader reader = ParquetFileReader.open(ParquetIO.file(file))) {
-      return footerMetrics(reader.getFooter(), metricsConfig, nameMapping);
+      return footerMetrics(reader.getFooter(), Stream.empty(), null, metricsConfig, nameMapping);
```

Comment:
This has the similar problem I mentioned in pr description for importing spark table; if the file itself is directly passed in there's not much chance to get the additional metrics tracked by value writers. Currently `fileMetrics` are only used by tests. Do people have suggestions on this?

Reply:
In that case, we should just set the NaN count map to null. I don't think that we want to scan imported files to create these metrics. Also, I believe that we can rely on recent Parquet versions to not produce min or max values that are NaN, so it should be safe to use these as long as we check that they are not NaN.

Reply:
Thank you! I'll leave this as `Stream.empty` for now.

Regarding relying on recent Parquet versions to not produce min or max values that are NaN, sounds like that also answers my question in the pr description (i.e. we won't follow this approach to populate upper and lower bounds). In my current code base it looks like parquet still gives us NaN as max; do you happen to have a reference to the parquet version that supports NaN properly? From a quick search I wasn't able to find it; I noticed [this](https://github.com/apache/parquet-mr/pull/515) but I suspect it's for something else. I'll look into it more deeply if you don't have it handy. 

> so it should be safe to use these as long as we check that they are not NaN.

Sorry to make sure I understand this correctly, sounds like only the three following cases will be valid:
1) v1 table, no NaN counter, min/max could have NaN - use the existing logic, we can't do much about min/max==NaN
2) v2 table, NaN counter exists, min/max will not be NaN - in this case metrics are produced by iceberg writer
3) v2 table, no NaN counter, min/max will not be NaN - in this case the file is imported or from this `fileMetrics`

Then to accommodate for (3) we will have to remember in evaluators that absence of NaN counter doesn't necessarily mean there's no NaN value in the column; but that might be fine since we will need this logic to accommodate (1) as well (unless we implement evaluators in a way that we can differentiate v1/2 table; not sure if we want that).

Reply:
I don't have a reference for the Parquet fix. I think I was in a sync where it was discussed. Maybe we should generate our own lower/upper bounds for Parquet then.

`Float.compare` will sort NaN values last, so if we do get max=NaN from Parquet our evaluators should still work as expected. It will just include a much larger range than necessary. If we can generate better stats for table metadata, then that would be ideal.

Your cases look correct, except that I would say that we expect only max=NaN or min=max=NaN from Parquet. Using the existing logic should be okay.

I agree that the lack of a NaN counter means that the value is unknown. This is the case for all files written to v1 tables. V2 tables will generally have the NaN counter values, but not in all cases (imported files).

Reply:
Thank you! I guess I'll revisit min/max status for Parquet after NaN related code are mostly done.